src/Enzyme.jl

```diff
@@ -916,9 +916,15 @@ grad = gradient(Reverse, f, [2.0, 3.0])
 ```
 """
 @inline function gradient(::ReverseMode, f, x)
-    dx = zero(x)
-    autodiff(Reverse, f, Active, Duplicated(x, dx))
-    dx
+    if Enzyme.Compiler.active_reg(typeof(x))
```

Comment:
Minor but I'm realizing the better check would be (saves compile time)

```julia
if active_reg_inner(Core.Typeof(f), #=seen=#(), #=world=#nothing, #=justActive=#Val(true)) == ActiveState
```

Reply:
I made this change, and it has the following effect, making my mixed activity namedtuple take the Ref path. I hope that sounds OK, the answer certainly looks right:
```julia
julia> xp2 = (x = [1.0, 2.0], p = 3.0);

julia> Enzyme.Compiler.active_reg(typeof(xp2))
ERROR: AssertionError: @NamedTuple{x::Vector{Float64}, p::Float64} has mixed internal activity types. See https://enzyme.mit.edu/julia/dev/#Mixed-Activity for more information

julia> Enzyme.Compiler.active_reg_inner(typeof(xp2), #=seen=#(), #=world=#nothing, #=justActive=#Val(true)) == Enzyme.Compiler.ActiveState
true

julia> gradient(Reverse, z -> sum(z.x .^ z.p), xp2)  # takes the Ref path
(x = [3.0, 12.0], p = 5.545177444479562)

julia> Zygote.gradient(z -> sum(z.x .^ z.p), xp2)[1]
(x = [3.0, 12.0], p = 5.545177444479562)
```
Maybe that needs a test.

Reply:
Yes it should be fine